### PR TITLE
CLOUDP-58264: Add Simple Replica Set Test

### PR DESCRIPTION
**Pull request by chatton**
### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

This PR creates a simple test using the operator-sdk test framework which ensures that the operator creates the ConfigMap and StatefulSet.

The go.mod was updated to work with the testing framework but will be reverted as soon as possible.

### CODE REVIEW

**Comment by LouisPlisso** [comment]
@chatton can you please check why the unit tests are failing: it looks like a gopath issue

File: go.mod [hunk 1]
```diff
@@ -3,13 +3,46 @@ module github.com/mongodb/mongodb-kubernetes-operator
...
 	github.com/stretchr/testify v1.4.0
 	go.uber.org/zap v1.13.0
-	k8s.io/api v0.15.9
-	k8s.io/apimachinery v0.15.9
+	golang.org/x/net v0.0.0-20191028085509-fe3aa8a45271
+	k8s.io/api v0.17.2
+	k8s.io/apimachinery v0.17.2
 	k8s.io/client-go v12.0.0+incompatible
-	sigs.k8s.io/controller-runtime v0.3.0
+	sigs.k8s.io/controller-runtime v0.5.0
```
Comment (LouisPlisso) [inline go.mod]: is this intentional?

we should stick to 1.16 max
also client-go is now following k8s versioning

File: go.mod [hunk 2]
```diff
@@ -3,13 +3,46 @@ module github.com/mongodb/mongodb-kubernetes-operator
...
 )
+
+// TODO: These replace statements need to be completely removed. These exist as the operator-sdk currently requires
+// them. Once they are fully removed, we should only include "require" statements in go.mod
```
Comment (LouisPlisso) [inline go.mod]:
```suggestion
// them. Once they are fully removed, we should not include "require" statements in go.mod
```

File: pkg/controller/mongodb/mongodb_controller.go [hunk 3]
```diff
@@ -26,7 +26,7 @@ import (
 )
 
 const (
-	automationConfigKey   = "automation-config"
+	AutomationConfigKey   = "automation-config"
 	agentName             = "mongodb-agent"
 	mongodbName           = "mongod"
 	agentImageEnvVariable = "AGENT_IMAGE"
```
Comment (LouisPlisso) [inline pkg/controller/mongodb/mongodb_controller.go]: we need to be consistent: either all capitalised or none, no?

File: scripts/ci/download.go [hunk 4]
```diff
@@ -10,7 +10,7 @@ import (
 
 // download.go uses the following environment variables:
 //   URL: The url of the file to download
-//   BINDIR: The directory which the newly downloaded file will be placed
+//   DIR: The directory which the newly downloaded file will be placed
 //   FILENAME: The name the file should have after being downloaded
```
Comment (LouisPlisso) [inline scripts/ci/download.go]: why are we using environment variables: argument passing is much more explicit and you can always pass env variables as argument if you really really need it

File: scripts/ci/generate_k8.sh [hunk 5]
```diff
@@ -6,4 +6,4 @@ set -xeo pipefail
 # GOROOT needs to be explicitly set otherwise we get the error
 # time="2020-02-24T17:52:08Z" level=info msg="Running deepcopy code-generation for Custom Resource group versions: [mongodb:[v1], ]\n"
 # deepcopy.go:885] Hit an unsupported type invalid type for invalid type, from ./pkg/apis/mongodb/v1.MongoDB
-GOROOT="/opt/golang/go1.13" operator-sdk generate k8s
+operator-sdk generate k8s
```
Comment (LouisPlisso) [inline scripts/ci/generate_k8.sh]: please adjust the comment if GOROOT is no more needed

File: pkg/controller/mongodb/mongodb_controller.go (hunk 3 above)
Comment (chatton) [reply pkg/controller/mongodb/mongodb_controller.go]: this one in particular is being imported in the e2e test.

**Pull request closed without merging**